iModelCore/ECPresentation/Source/Content/ContentSpecificationsHandler.cpp

```diff
@@ -226,18 +256,85 @@ static ECSchemaHelper::RelationshipPathsResponse FindRelatedPropertyPaths(Conten
             }
         }
+    return specs;
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+static bvector<std::unique_ptr<FlattenedRelatedPropertiesSpecification>>CreateForNestedPropertiesFromModifiers(bvector<std::unique_ptr<FlattenedRelatedPropertiesSpecification>> const& flatSpecs, bvector<ContentModifierCP> const& modifiers, ECSchemaHelper const& helper);
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+static bvector<std::unique_ptr<FlattenedRelatedPropertiesSpecification>>CreateForFlatSpecFromModifiers(RelatedPropertiesSpecificationCR spec, bvector<ContentModifierCP> modifiers, ECSchemaHelper const& helper)
+    {
+    bvector<std::unique_ptr<FlattenedRelatedPropertiesSpecification>> specs;
+    if (!spec.AllPropertiesIncluded())
+        return specs;
+
+    int iter = -1;
+    while (iter != 0)
+        {
+        iter = 0;
+        for (ContentModifierCP& modifier : modifiers)
+            {
+            if (modifier == nullptr || !modifier->ShouldApplyOnNestedContent())
+                continue;
+
+            ECRelationshipConstraintClassList const* classes = GetClassFromRelationship(helper, spec);
+            for (ECClassCP ecClass : *classes)
+                {
+                if (!ecClass->Is(modifier->GetSchemaName().c_str(), modifier->GetClassName().c_str()))
+                    continue;
+                DiagnosticsHelpers::ReportRule(*modifier);
+                bvector<std::unique_ptr<FlattenedRelatedPropertiesSpecification>> createdSpecs = FlattenedRelatedPropertiesSpecification::Create(modifier->GetRelatedProperties(), RelatedPropertiesSpecificationScopeInfo(modifier->GetPropertyCategories()));
+                if (createdSpecs.empty())
+                    continue;
+
+                modifier = nullptr;
+                ++iter;
+                FlattenedRelatedPropertiesSpecification::MoveNestedSpecification(specs, std::move(createdSpecs), spec);
+                ContainerHelpers::MovePush(specs, CreateForNestedPropertiesFromModifiers(specs, modifiers, helper));
+                }
+            }
+        }
+    return specs;
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+static bvector<std::unique_ptr<FlattenedRelatedPropertiesSpecification>>CreateForNestedPropertiesFromModifiers(bvector<std::unique_ptr<FlattenedRelatedPropertiesSpecification>> const& flatSpecs, bvector<ContentModifierCP> const& modifiers, ECSchemaHelper const& helper)

```

Comment:
missing space between return type and function name

Reply:
fixed

Reply:
fixed